cloudinary/api_client/execute_request.py

```diff
@@ -0,0 +1,55 @@
+import json
+import socket
+
+from urllib3.exceptions import HTTPError
+
+from cloudinary.exceptions import (
+    BadRequest,
+    AuthorizationRequired,
+    NotAllowed,
+    NotFound,
+    AlreadyExists,
+    RateLimited,
+    GeneralError
+)
+from cloudinary.utils import process_params
+
+EXCEPTION_CODES = {
+    400: BadRequest,
+    401: AuthorizationRequired,
+    403: NotAllowed,
+    404: NotFound,
+    409: AlreadyExists,
+    420: RateLimited,
+    500: GeneralError
+}
+
+
+class Response(dict):
+    def __init__(self, result, response, **kwargs):
+        super(Response, self).__init__(**kwargs)
+        self.update(result)
+
```

Comment:
Where is all rate-limit related handling? `self.rate_limit_allowed` and others?

Reply:
@const-cloudinary 

Rate limit data was removed from the response in the reference implementation (`cloudinary_npm`) as part of this feature (see https://github.com/cloudinary/cloudinary_npm/commit/740590c9d332e3885eb67b090ec8fe4a26ffb9d6).

I just confirmed that `rate_limit_.*` existed in `cloudinary_npm` admin responses before this commit but is no longer a part of the response in the latest master.

I am assuming this is an unintentional regression in `cloudinary_npm`? We can open a bug on `cloudinary_npm` and fix this (including adding testing).

Reply:
@const-cloudinary Following slack conversation, we will bring back rate_limits and update the PR